mllib/src/main/scala/org/apache/spark/ml/tuning/CrossValidator.scala

```diff
@@ -100,31 +104,44 @@ class CrossValidator @Since("1.2.0") (@Since("1.4.0") override val uid: String)
     val eval = $(evaluator)
     val epm = $(estimatorParamMaps)
     val numModels = epm.length
-    val metrics = new Array[Double](epm.length)
+    val numPar = $(numParallelEval)
 
     val instr = Instrumentation.create(this, dataset)
     instr.logParams(numFolds, seed)
     logTuningParams(instr)
 
+    // Compute metrics for each model over each fold
```

Comment:
technically it's each `estimator` rather than `model` - since both the fitting (estimator or pipeline) and evaluation (model or pipeline model) is in parallel.